### PR TITLE
MCOL-4910: Turning on C++ 20 for Columnstore

### PR DESCRIPTION
**Pull request by mariadb-LeonidFedorov**
Turning on C++ 20
No diffs in build environment except
Ubuntu 18.04 support removed
- volatile keyword removed
- some warnings suppressed 
- STLAllocator changed for new std::allocator signatures
- boost::Format removed and replaced with dummy handwritten function
- boost::RegEx replaced with std::regex with handmade extention
- boost::property_tree and boost::json_parsed replaced with json header only parser
[nlochman ](https://github.com/nlohmann/json) not fastest, but very useful, simple and good tested
[see benchmark](https://github.com/miloyip/nativejson-benchmark/blob/master/sample/performance_Corei7-4980HQ@2.80GHz_mac64_clang7.0_1._Parse_Memory_(byte).png)
Can be switched to [new boost.json](http://vinniefalco.github.io/doc/json/) 
Or [simjson](https://github.com/simdjson/simdjson) when we need perfomance

PR is passing test-001

PS. There is ```if constexpr``` here in code

### CODE REVIEW

File: utils/loggingcpp/format.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/*
+   Copyright (C) 2014 InfiniDB, Inc.
```
Comment (drrtuy) [inline utils/loggingcpp/format.h]: I doubt that InfiniDB adds something to this file except maybe a header.

File: utils/loggingcpp/format.h [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            throw std::logic_error("logggin::format: unexpected type in argslist");
+        }
+        ++iter;
+        ++position;
+    }
+    static std::regex restToken("%[0-9]%");
+    errMsg = std::regex_replace(errMsg, restToken, "");
+}
+
+} // namespace logging
```
Comment (drrtuy) [inline utils/loggingcpp/format.h]: Can you follow the style that you suggested?
I would appreciate if you put vim trailer here also.

**Review by drrtuy — COMMENTED** [review]
Could you ask Daniel to compare the performance of this build with his latest measurements of develop-6 before we merge?

File: utils/loggingcpp/format.h (hunk 2 above)
Comment (mariadb-LeonidFedorov) [reply utils/loggingcpp/format.h]: I have formatted new file like this, others will be better to format once in one big PR

File: utils/loggingcpp/format.h (hunk 1 above)
Comment (mariadb-LeonidFedorov) [reply utils/loggingcpp/format.h]: :)

**Pull request closed without merging**